internal/worker/server.go

```diff
@@ -651,7 +651,7 @@ func (s *Server) JobArtifactLocation(id uuid.UUID, name string) (string, error)
 
 	p := path.Join(s.config.ArtifactsDir, id.String(), name)
 	if _, err := os.Stat(p); errors.Is(err, os.ErrNotExist) {
-		return "", fmt.Errorf("Artifact not found: %s", p)
+		return p, fmt.Errorf("Artifact not found: %s", p)
```

Comment:
The `p` as the path that was checked is part of the error message. I'd say that the correct way would be to return a structured error that has the path stored in a property, so that it can be accessed.

Reply:
This is on purpose so that when the file isn't present the path that it is expected to be at can be used to write it in `TestDownload`, this is much cleaner than creating a new error structure.

Reply:
Although I disagree on what is cleaner, I can accept the current version. Defining a custom error struct seems easy and clean to me. :shrug: Especially if the only use case is in tests.

If you keep it as is, it would be good to note the meaning of the returned value in the function doc text in case an error is also returned.